base/regex.jl

```diff
@@ -364,12 +364,18 @@ function endswith(s::SubString{String}, r::Regex)
 end
 
 function chopprefix(s::AbstractString, prefix::Regex)
+    # fast path to avoid (some, letter-only) regex
+    all(isletter, prefix.pattern) && startswith(s, prefix.pattern) && return chopprefix(s, prefix.pattern)
```

Comment:
Why would a user use a regex in this case?

Reply:
For case-insensitivity a regex is (currently) the only option. But I check first for exact match to avoid using PCRE.

Reply:
It's over 3x faster, when fast-path taken, so what's the objection/downvote referring to?